controller: Fix incorrect ports name matching when validating ports

liangwen12year opened the pull request:
Resolves: https://issues.redhat.com/browse/RHEL-1414

liangwen12year committed:
controller: Fix incorrect ports name matching when validating ports

When applying the following desired state, the NmstateError
"InvalidArgument: Interface bond0 has controller br0 but not listed in
port list of controller interface" will be raised.

The root cause is that the function `validate_controller_not_in_port_list()`
validates if the controller name of the merged interface is in the port
list of the controller interface. This is incorrect, instead, we should
validate if the name of merged interface itself appears in the port
list of the controller.

```
- name: bond0
  type: bond
  state: up
  link-aggregation:
    mode: balance-rr
  controller: br0
- name: br0
  type: ovs-bridge
  state: up
  bridge:
    port:
      - name: bond0
```

Resolves: https://issues.redhat.com/browse/RHEL-1414

Signed-off-by: Wen Liang <liangwen12year@gmail.com>

cathay4t reviewed (changes requested):
Please do not modify existing test case, they might be for other use case.

Meanwhile, commit message should explains:
 * What's the problem
 * Root cause
 * How did you fix it.

liangwen12year commented:
> Please do not modify existing test case, they might be for other use case.
> 
> Meanwhile, commit message should explains:
> 
> * What's the problem
> * Root cause
> * How did you fix it.

I added the unit test case and modified the commit message, thanks.

cathay4t reviewed (approved):
LGTM. Thanks!
